python/ray/serve/_private/router.py

```diff
@@ -121,6 +127,15 @@ async def assign_replica(
         )
 
     def update_running_replicas(self, running_replicas: List[RunningReplicaInfo]):
+        new_replica_ids = {r.replica_tag for r in running_replicas}
+        if new_replica_ids != self._replica_id_set:
```

Comment:
why do we get the replica update without change? I am not sure why we have to check here.

Reply:
this is purely for logging

Reply:
we might get spurious updates or the update might only be a config change